x/crosschain/keeper/keeper_cross_chain_tx_vote_outbound_tx.go

```diff
@@ -105,32 +103,20 @@ func (k msgServer) VoteOnObservedOutboundTx(goCtx context.Context, msg *types.Ms
 	// FinalizeOutbound sets final status for a successful vote
 	// FinalizeOutbound updates CCTX Prices and Nonce for a revert
 
-	err = func() error { //err = FinalizeOutbound(k, ctx, &cctx, msg, ballot.BallotStatus) // remove this line
+	tmpCtx, commit := ctx.CacheContext()
+	err = func() error { //err = FinalizeOutbound(k, ctx, &cctx, msg, ballot.BallotStatus)
```

Comment:
Inline the function FinalizeOutbound() as it's not reused anywhere else, and as a standalone function it relies heavily on side effects. As such inlining it is better by reducing one layer of indirection-- the parameters.